src/coreComponents/mesh/generators/InternalWellGenerator.cpp

```diff
@@ -360,10 +370,15 @@ void InternalWellGenerator::connectPerforationsToWellElements()
     real64 const wellLength = m_nodeDistFromHead[m_elemToNodesMap[iwelemBottom][LineBlockABC::NodeLocation::BOTTOM]];
 
     GEOS_THROW_IF( m_perfDistFromHead[iperf] > wellLength,
-                   "Distance from perforation " << perf.getName() << " to head is larger than well polyline length for well " << getName() << "\n \n"
-                                                << "Here is how the \"distanceFromHead\" keyword is used in the definition of the perforation location: \n"
-                                                << "We start from the well head (top of the well) and we measure the linear distance along the well polyline as we go down the well.\n"
-                                                << "When we reach the distanceFromHead specified by the user, we place a perforation on the well at this location of the polyline, and connect it to the reservoir element that contains this perforation",
+                   GEOS_FMT( "{}: Distance from well perforation to head ({} = {}) is larger than well"
+                             " polyline length ({})\n \n You should check the following values:"
+                             "\n   1 - {}\n   2 - {}, Z values",
+                             perf.getWrapperDataContext( Perforation::viewKeyStruct::distanceFromHeadString() ),
+                             Perforation::viewKeyStruct::distanceFromHeadString(),
+                             m_perfDistFromHead[iperf], wellLength,
+                             perf.getWrapperDataContext( Perforation::viewKeyStruct::distanceFromHeadString() ),
+                             WellControls::viewKeyStruct::refElevString(), m_wellControlsName,
+                             getWrapperDataContext( viewKeyStruct::polylineNodeCoordsString() ) ),
```

Comment:
Unless there a drawback, can you do back to `<<` the new code is complicated since the flow is broken by `fmt`.